src/eos-updater-fetch.c

```diff
@@ -295,18 +299,451 @@ content_fetch_old (EosUpdater      *updater,
   return TRUE;
 }
 
-static void
-content_fetch (GTask *task,
-               gpointer object,
-               gpointer task_data,
-               GCancellable *cancellable)
+
+static GFile *
+get_temporary_directory_to_checkout_in (GError **error)
 {
-  EosUpdater *updater = EOS_UPDATER (object);
-  EosUpdaterData *data = task_data;
-  g_autoptr(GError) error = NULL;
-  g_autoptr(GMainContext) task_context = g_main_context_new ();
+  g_autofree gchar *temp_dir = g_dir_make_tmp ("ostree-checkout-XXXXXX", error);
+  g_autofree gchar *path = NULL;
 
-  g_main_context_push_thread_default (task_context);
+  if (!temp_dir)
+    return NULL;
+
+  path = g_build_filename (temp_dir, "checkout", NULL);
+  return g_file_new_for_path (path);
+}
+
+static GFile *
+inspect_directory_in_ostree_repo (OstreeRepo    *repo,
```

Comment:
I'll move it out into a helper library, but I'm not sure how we will unit test something like this without essentially doing what the acceptance tests are already doing (I don't think it's possible to make an in-memory `OstreeRepo`, is it?)

Reply:
It’s not possible to have an in-memory `OstreeRepo`. I think in this case the main advantage comes from being able to expose more internal state which you can check in the test code, rather than essentially summarising _everything_ in the exit status of a program.

If possible, it would be good to eliminate some environment variables this way. Environment variables are almost always a hack to expose/change internal state from somewhere it shouldn’t be exposed.

Reply:
Eliminating the environment variables would be nice, but I don't think we can do that without the expense of not being able to do end-to-end testing - they're kind of a necessary evil for that.

(I've refactored things a little so that at least for the parts we expose to the unit tests, we're not reliant on environment variables).